linera-execution/src/execution.rs

```diff
@@ -251,74 +248,6 @@ where
         Ok(())
     }
 
-    /// Schedules application registration messages when needed.
-    ///
-    /// Ensures that the outgoing messages in `results` are preceded by a system message that
-    /// registers the application that will handle the messages.
-    pub async fn update_execution_outcomes_with_app_registrations(
-        &self,
-        txn_tracker: &mut TransactionTracker,
-    ) -> Result<(), ExecutionError> {
-        let results = txn_tracker.outcomes_mut();
-        let user_application_outcomes = results.iter().filter_map(|outcome| match outcome {
-            ExecutionOutcome::User(application_id, result) => Some((application_id, result)),
-            _ => None,
-        });
-
-        let mut applications_to_register_per_destination = BTreeMap::<_, BTreeSet<_>>::new();
-
-        for (application_id, result) in user_application_outcomes {
-            for message in &result.messages {
-                applications_to_register_per_destination
-                    .entry(&message.destination)
-                    .or_default()
-                    .insert(*application_id);
-            }
-        }
-
-        if applications_to_register_per_destination.is_empty() {
-            return Ok(());
-        }
-
-        let messages = applications_to_register_per_destination
-            .into_iter()
-            .map(|(destination, applications_to_describe)| async {
-                let applications = self
-                    .system
-                    .registry
-                    .describe_applications_with_dependencies(
-                        applications_to_describe.into_iter().collect(),
-                    )
-                    .await?;
-
-                Ok::<_, ExecutionError>(RawOutgoingMessage {
-                    destination: destination.clone(),
-                    authenticated: false,
-                    grant: Amount::ZERO,
-                    kind: MessageKind::Simple,
-                    message: SystemMessage::RegisterApplications { applications },
-                })
-            })
-            .collect::<FuturesOrdered<_>>()
-            .try_collect::<Vec<_>>()
-            .await?;
-
-        let system_outcome = RawExecutionOutcome {
-            messages,
-            ..RawExecutionOutcome::default()
-        };
-
-        // Insert the message before the first user outcome.
-        let index = results
-            .iter()
-            .position(|outcome| matches!(outcome, ExecutionOutcome::User(_, _)))
-            .unwrap_or(results.len());
-        // TODO(#2362): This inserts messages in front of existing ones, invalidating their IDs.
-        results.insert(index, ExecutionOutcome::System(system_outcome));
```

Comment:
Awesome! :tada: 
Don't forget to add https://github.com/linera-io/linera-protocol/issues/2362 to the issues that will be closed by this PR.